### ISSUE TITLE
Include inventory file for interlinking docs

### ISSUE THREAD

**Issue by sethaxen** [LuxDL/DocumenterVitepress.jl#189]
From v1.3.0, Documenter has included an `objects.inv` inventory file in the built docs, which packages to use [DocumenterInterLinks.jl](https://github.com/JuliaDocs/DocumenterInterLinks.jl) to link to documentation of methods/objects in other packages. This works fine for Documenter-built packages, but DocumenterVitepress does not seem to include an inventory file. It would be great if this package could include this file.

**Comment by goerz** [comment]
Semi-related: Does the outgoing direction work? That is, can [DocumenterInterLinks](https://github.com/JuliaDocs/DocumenterInterLinks.jl) be used in Vitepress sites to link out to other projects that do have an inventory? I think `DocumenterVitepress` doesn't touch those parts of the `Documenter` pipeline, so I'm guessing (hoping) that the plugin still works. It would be nice to have confirmation, though. Same for [DocumenterCitations](https://github.com/JuliaDocs/DocumenterCitations.jl): Do citations in Vitepress sites still work?

**Comment by asinghvi17** [comment]
Hey, yeah I had been looking at it as part of a full refactor to get tools from here into DocumenterMarkdown.  I still have that code and not too much has changed, so I might look at just including that as a subdir here for the time being.

One thing that may prove difficult is reverse-engineering Vitepress link structures, I'm not 100% sure how much they differ from Documenter's structures.  We may not get everything exactly right while writing this inventory.

But DocumenterInterLinks works fine, most Documenter extensions that go back to MarkdownAST (and don't override renderer methods) should just work.

**Comment by goerz** [comment]
> One thing that may prove difficult is reverse-engineering Vitepress link structures, I'm not 100% sure how much they differ from Documenter's structures.

Ok, so I had a look at the source code. I don't think I'd try to override the `_get_inventory_uri` methods from `Documenter`. Instead, I'd probably put the [inventory writing code](https://github.com/JuliaDocs/Documenter.jl/blob/master/src/html/write_inventory.jl) directly in the DocumenterVitepress' [main `render` function](https://github.com/LuxDL/DocumenterVitepress.jl/blob/c2fc02e4fab6d819c63cbac22bbfb0994c0b4486/src/writer.jl#L112). The recursive `render` methods for the various markdown elements contain the code to calculate their respective anchor id's, e.g., [for docstrings](https://github.com/LuxDL/DocumenterVitepress.jl/blob/c2fc02e4fab6d819c63cbac22bbfb0994c0b4486/src/writer.jl#L383). These render methods also get and forward `kwargs...`. So what I'm thinking: set up a data structure to collect the inventory item data in the [top-level `render` method](https://github.com/LuxDL/DocumenterVitepress.jl/blob/c2fc02e4fab6d819c63cbac22bbfb0994c0b4486/src/writer.jl#L112) and pass it as a keyword argument to the downstream `render` methods, which can then store the relevant data in that structure. Then, at the end, the main `render` method can write out the actual `objects.inv` file based on the collected data.

**Comment by asinghvi17** [comment]
Makes sense!  And we can then mark anchors etc. as well.

**Comment by goerz** [comment]
How do you feel about adding [DocInventories.jl](https://github.com/JuliaDocs/DocInventories.jl) as a dependency? If you want to be extremely conservative with dependencies, it's possible to implement the inventory writing without `DocInventories` (by duplicating the code from the rather minimal parts of `DocInventories` that are actually used; as we did in `Documenter`). But, `DocInventories` is also not a heavy dependency, with `CodecZlib` as its only non-stdlib indirect dependency, and we definitely need `CodecZlib` in `DocumenterVitepress` to write out the inventory. So adding `DocInventories` as a dependency in `DocumenterVitepress` would probably be better, and keep the code simpler. Let me know if that's okay.

**Comment by asinghvi17** [comment]
I've got nothing against it - we depend on Node of all things so this is not a dependency shy library :D

I'm assuming that DocumenterInterLinks uses that package?  My only concern is that it shouldn't go out of sync, but if it's used by DocumenterInterLinks then presumably that format is staying the same.

**Comment by goerz** [comment]
Yes, `DocInventories` is a dependency of `DocumenterInterLinks`. Also, `DocInventories` should be extremely stable. I don't really expect it to need any changes in the future. Maybe some new features for interactively exploring inventories in rich environments like Pluto, but nothing that would affect the I/O or break compatibility. And the format itself is set by Sphinx, and hasn't changed in the 15 years since it was developed :-)

**Comment by goerz** [comment]
I started playing around with this in #249
